jsonnet/main.jsonnet

```diff
@@ -320,7 +320,30 @@ local inCluster =
     },
 
     // Objects
-    clusterMonitoringOperator: clusterMonitoringOperator($.values.clusterMonitoringOperator),
+    clusterMonitoringOperator: clusterMonitoringOperator($.values.clusterMonitoringOperator) {
+      // The cluster-monitoring-operator ClusterRole needs the combined set of
+      // permissions from all its operand ClusterRoles.  This extends the base
+      // ClusterRole by just appending the rules from the others.
+      clusterRole+: {
+        rules+: inCluster.alertmanager.clusterRole.rules +
+                inCluster.clusterMonitoringOperator.clusterRoleView.rules +
+                inCluster.clusterMonitoringOperator.userWorkloadConfigEditRole.rules +
+                inCluster.grafana.clusterRole.rules +
+                inCluster.kubeStateMetrics.clusterRole.rules +
+                inCluster.nodeExporter.clusterRole.rules +
+                inCluster.openshiftStateMetrics.clusterRole.rules +
+                inCluster.prometheusAdapter.clusterRole.rules +
+                inCluster.prometheusAdapter.clusterRoleAggregatedMetricsReader.rules +
+                inCluster.prometheusAdapter.clusterRoleServerResources.rules +
+                inCluster.prometheus.clusterRole.rules +
+                [role.rules for role in inCluster.prometheus.roleSpecificNamespaces.items] +
```

Comment:
No need for a loop as all those rules are the same ones. More in https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/components/prometheus.libsonnet#L215-L246

Reply:
Ack. I dunno, seems fine and future proof though. Is the alternative to just reach into the items array and grab only the first item?